File: src/phorge/api/endpoints/environment.py
"""Environment file API endpoints."""

from __future__ import annotations

from phorge.api.client import ForgeClient


class EnvironmentAPI:
    def __init__(self, client: ForgeClient) -> None:
        self._client = client

    async def get(self, server_id: int, site_id: int) -> str:
        return await self._client.get_text(
            f"/servers/{server_id}/sites/{site_id}/env"
        )

    async def update(self, server_id: int, site_id: int, content: str) -> None:
        await self._client.put(
            f"/servers/{server_id}/sites/{site_id}/env",
            json={"content": content},
        )

File: internal/forge/environment.go
package forge

import (
	"context"
	"net/http"
)

// envContent wraps the "content" attribute of the environment JSON:API resource.
type envContent struct {
	Content string `json:"content"`
}

// Get returns the environment file contents for a site.
func (s *EnvironmentService) Get(ctx context.Context, serverID, siteID int64) (string, error) {
	out, err := getResource[envContent](s.client, ctx, s.client.orgPath("/servers/%d/sites/%d/environment", serverID, siteID))
	if err != nil {
		return "", err
	}
	return out.Content, nil
}

// Update replaces the environment file contents for a site.
func (s *EnvironmentService) Update(ctx context.Context, serverID, siteID int64, content string) error {
	path := s.client.orgPath("/servers/%d/sites/%d/environment", serverID, siteID)
	body := map[string]string{"content": content}
	return s.client.do(ctx, http.MethodPut, path, body, nil)
}
